### PR TITLE
Discuss new meta- and sub-repos in user guide

### PR DESCRIPTION
**Pull request by bpeabody**
addresses https://github.com/twosigma/git-meta/issues/202

### CODE REVIEW

File: doc/user-guide.md [hunk 1]
```diff
@@ -291,6 +296,23 @@ does exactly the right thing:
 $ git clone http://example.com/your-meta-repo.git meta
 ```
 
+### Creating a new sub-repository
+
+Assuming that you are using the
+[Omega Repository](https://github.com/twosigma/git-meta/wiki/The-Omega-Repo)
```
Comment (novalis) [inline doc/user-guide.md]: relative link here?

Comment (bpeabody) [inline doc/user-guide.md]: It's currently in the wiki, not the repo itself. Maybe I should move it?

Comment (bpeabody) [inline doc/user-guide.md]: Yeah, I'll fold it into the architecture document itself.

**Comment by bpeabody** [comment]
OK, I took a stab at reworking the architecture document to be inline with our latest omega strategy.

**Comment by bpeabody** [comment]
@novalis ping

**Comment by bpeabody** [comment]
@jhedwardyang @novalis either of you want to sign off on this? It's just doc changes, but I'd like to get it in.

**Comment by novalis** [comment]
I'll take a look.

File: doc/architecture.md [hunk 2]
```diff
@@ -39,11 +39,12 @@ space.  In short, the first section should explain why git-meta is needed.
 
 Next, we present the architecture for implementing a mono-repo using Git
 submodules.  We describe the overall repository structure, commits, forking,
-refs, and the client-side representation.
+refs, client-side representation, and a recommended server-side configuration.
 
-Then, we present an earlier, more-intuitive architecture.  Seeing how our
-strategy evolved from this approach is illustrative, and helps to understand
-some of the less-intuitive design choices.
+Then, we discuss the how our current design evolved from seemingly simple goal
```
Comment (novalis) [inline doc/architecture.md]: "the how"

Comment (novalis) [inline doc/architecture.md]: "from seemingly simple goal"

File: doc/architecture.md [hunk 3]
```diff
@@ -539,13 +530,15 @@ by the new synthetic-meta-ref if it does not already contain that commit in its
...
 
-# Naive Architecture
+# Design and Evolution
 
-In this section we describe the basic model we had in mind when we started
-git-meta.  First, we provide an overview of the original architecture.  Then,
-we describe a series of problems that arise from this architecture.  Finally,
-we draw conclusions by this exercise and connect them to our final design
-choices.
+In this section we show how the architecture of git-meta evolved to better
```
Comment (novalis) [inline doc/architecture.md]: s/evolved to/evolved, in order to/

**Pull request merged**